src/rpc/mining.cpp

```diff
@@ -344,8 +344,10 @@ UniValue getmininginfo(const UniValue& params, bool fHelp)
 
     UniValue obj(UniValue::VOBJ);
     obj.pushKV("blocks",           (int)chainActive.Height());
-    obj.pushKV("currentblocksize", (uint64_t)nLastBlockSize);
-    obj.pushKV("currentblocktx",   (uint64_t)nLastBlockTx);
+    if (nLastBlockSize > 0) {
+        obj.pushKV("currentblocksize", (uint64_t)nLastBlockSize);
+        obj.pushKV("currentblocktx",   (uint64_t)nLastBlockTx); // not including coinbase
+    }
```

Comment:
Personally, I prefer the `std::optional` approach that upstream took, because it makes semantic sense (these don't have a value until `CreateNewBlock` is called). We don't need to use their `Optional` type (which they introduced in order to start migrating to using optionals), since we started from C++11 and have always had `std::optional`.

Given that, we should just follow what upstream commit does, and change the names of these statics etc.